airbyte-integrations/base-python/base.py

```diff
@@ -66,25 +70,30 @@ def start(self):
                 sys.exit(0)
 
             rendered_config_path = os.path.join(temp_dir, 'config.json')
-            config_object = source.read_config(parsed_args.config)
-            source.render_config(config_object, rendered_config_path)
+            raw_config = source.read_config(parsed_args.config)
+            rendered_config = source.transform_config(raw_config)
+            source.write_config(rendered_config, rendered_config_path)
+
+            config_container = ConfigContainer(
+                raw_config=raw_config,
+                rendered_config=rendered_config,
+                raw_config_path=parsed_args.config,
+                rendered_config_path=rendered_config_path)
 
-            # todo: output message for check
             if cmd == "check":
-                check_result = source.check(logging, rendered_config_path)
+                check_result = source.check(log_line, config_container)
                 if check_result.successful:
-                    print("Check succeeded")
+                    log_line("Check succeeded", "INFO")
                     sys.exit(0)
                 else:
-                    print("Check failed")
+                    log_line("Check failed", "ERROR")
                     sys.exit(1)
             elif cmd == "discover":
-                schema = source.discover(logging, rendered_config_path)
-                print(schema.schema)
+                catalog = source.discover(log_line, config_container)
+                print(catalog.serialize())
```

Comment:
Since we want to dictate the precise format of what we're outputting, we don't want an `slf4j`/logger equivalent.

It's debatable if we should use `sys.stdout.write()` or `print`, but `print` just seemed easier (we do want newlines after everthing, and we it's easy to use the string conversions in some cases).